zerver/lib/actions.py

```diff
@@ -4631,13 +4633,45 @@ def get_service_dicts_for_bots(user_profile_id: str) -> List[Dict[str, Any]]:
             pass
     return service_dicts
 
+def get_service_dicts_for_bots(user_profile_ids: List[str]) -> Dict[str, List[Dict[str, Any]]]:
+    bot_profiles = get_user_profiles_by_ids(user_profile_ids)
+    bot_services_by_uid = get_services_for_bots(user_profile_ids)
+    bot_configs = get_bot_configs(user_profile_ids)
+
+    service_dicts_by_uid = {}  # type: Dict[str, List[Dict[Text, Any]]]
+    logging.warning('bot_profiles: %s' % (len(bot_profiles)))
+    for bot_profile in bot_profiles:
+        services = bot_services_by_uid[bot_profile.id]
+        service_dicts = []
+        logging.warning('bot_type: %s=%s' % (bot_profile.id, bot_profile.bot_type))
+        if bot_profile.bot_type == UserProfile.OUTGOING_WEBHOOK_BOT:
+            service_dicts = [{'base_url': service.base_url,
+                            'interface': service.interface,
+                            }
+                            for service in services]
+        elif bot_profile.bot_type == UserProfile.EMBEDDED_BOT:
+            try:
+                if bot_profile.id in bot_configs.keys():
+                    bot_config = bot_configs[bot_profile.id]
+                    service_dicts = [{'config_data': bot_config,
+                                    'service_name': services[0].name
+                                    }]
+            # A ConfigError just means that there are no config entries for user_profile.
+            except ConfigError:
+                pass
```

Comment:
This try/except seems awkward; do you really need it or can we do a more precise check?